sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/query.sql

```diff
@@ -242,23 +243,21 @@ first_seen_date AS (
   GROUP BY
     client_id
 ),
--- The next CTE returns the second_seen_date. It finds the next date after first_seen_date
--- as reported by the main ping. Further dates reported by other pings are skipped.
--- The source ping for second_seen_date is always the main ping.
+-- The next CTE returns the second_seen_date calculated as the next date reported by the
+--  main ping after first_seen_date. Dates reported by other pings are excluded.
 second_seen_date AS (
   SELECT
     client_id,
-    IF(
-      ARRAY_LENGTH(ARRAY_AGG(seen_dates)) > 1,
-      ARRAY_AGG(seen_dates ORDER BY value_date ASC)[SAFE_OFFSET(1)],
-      NULL
-    ) AS second_seen_date
+    ARRAY_AGG(seen_dates ORDER BY value_date ASC)[SAFE_OFFSET(0)] AS second_seen_date
```

Comment:
If we say second date can only come from the main ping, wouldn't it be easier to use the `main_ping` CTE here as the source instead which already contains sorted dates field? This way we could avoid unnesting and aggregating again?

Reply:
Yes, I think you're right, now that we have condition `> first_seen_date` allows us to use the main ping aggregation. Since this data is already tested, are you happy that I open a Jira ticket as an improvement to implement together with the data checks?

Reply:
Indeed, two aggregations reduced, updated, thanks!